lib/libspl/include/os/freebsd/sys/ia32/asm_linkage.h

```diff
@@ -0,0 +1,178 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or https://opensource.org/licenses/CDDL-1.0.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
+ * Use is subject to license terms.
+ */
+
+#ifndef _IA32_SYS_ASM_LINKAGE_H
+#define	_IA32_SYS_ASM_LINKAGE_H
+
+#if defined(__linux__) && defined(CONFIG_SLS)
```

Comment:
This should be in `lib/libspl/include/os/linux/sys/ia32/asm_linkage.h`. This is both the wrong platform. It is also in userspace where `CONFIG_SLS` is never defined.

That being said, would it be terrible to do this unconditionally everywhere in the interest of preemptive security? It is just dead code and honestly, our binary sizes are not going to be significantly influenced by this.

Reply:
I copied the original (linux) over, so it certainly can be cleaned up for freebsd.  Hopefully the Linux versions are "untouched" beyond what I added.